api/client/client.go

```diff
@@ -3389,3 +3389,44 @@ func (c *Client) GetHeadlessAuthentication(ctx context.Context, id string) (*typ
 	}
 	return headlessAuthn, nil
 }
+
+func (c *Client) CreatePlugin(ctx context.Context, plugin types.Plugin) error {
+	p, ok := plugin.(*types.PluginV1)
+	if !ok {
+		return trace.BadParameter("unknown plugin type")
+	}
+	_, err := c.grpc.CreatePlugin(ctx, p, c.callOpts...)
+	if err != nil {
+		return trail.FromGRPC(err)
+	}
+	return nil
+}
+
+func (c *Client) DeleteAllPlugins(ctx context.Context) error {
+	return trace.NotImplemented(notImplementedMessage)
+}
+
+func (c *Client) DeletePlugin(ctx context.Context, name string) error {
+	_, err := c.grpc.DeletePlugin(ctx, &types.ResourceRequest{Name: name})
+	return trace.Wrap(err)
+}
+
+func (c *Client) GetPlugin(ctx context.Context, name string, withSecrets bool) (types.Plugin, error) {
+	return nil, trace.NotImplemented(notImplementedMessage)
```

Comment:
Should the client call the server and let the server return the NotImplemented error?